staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -320,7 +174,7 @@ func startCompactorOnce(c storagebackend.TransportConfig, interval time.Duration
 	compactorsMu.Lock()
 	defer compactorsMu.Unlock()
 
-	key := fmt.Sprintf("%v", c) // gives: {[server1 server2] keyFile certFile caFile}
+	key := fmt.Sprintf("%v", c) // gives: {[server1 server2] keyFile certFile caFile}  // TODO does this make sense?
```

Comment:
Can we just use `c` as the key now that we're creating one `TransportConfig` per override?

Reply:
Reading `startCompactorOnce` and `etcd3.StartCompactor` makes me think this whole thing is subtly broken today (the interval checking and compactor replacement does not line up with how duplicate endpoints are handled and creating a string key from a struct with functions has dangerous edges).  Not sure what the best way to fix this is.

Reply:
FWIW, I would feel a lot more comfortable if we just had something like:

```go
type TransportConfigKey struct {
	ServerList    string // strings.join() of the TransportConfig.ServerList
	KeyFile       string
	CertFile      string
	TrustedCAFile string
}
```

Then at least there is not question what the key is.